### ISSUE TITLE
modeladmin list_filter display options of how filter should function

### ISSUE THREAD

**Issue by onno-timmerman** [wagtail/wagtail#9977]
### Is your proposal related to a problem?

When you use modeladmin for example to show all your articles with particular filters then you can use list_filter(). The filter does show all filter option in nice button like way on the right side of wagtail admin. However if you have like 50 authors and want to create a filter option thats not very long it would be nice to have an option to show these as a select field. Even if the select allows you to only show it per one item.

### Describe the solution you'd like
Maybe if list filter would be able to edited as a dict in this form
list_filter = {'somefield': 'buttons', 'somefield2': 'select'}

### Describe alternatives you've considered
hack my way in the sidebar or create full custom view.

**Comment by gasman** [comment]
Thanks for the suggestion @onno-timmerman! We're limited in how much we can do here, because modeladmin's implementation of filters is built upon the one in django-admin, which has similar limitations.

Our plan over the next couple of releases is to enhance the capabilities of snippets to the point where they can fully replace modeladmin, and then phase out modeladmin. Snippets already provide the flexibility you're looking for, via the `filterset_class` option: https://docs.wagtail.org/en/stable/topics/snippets.html#customising-snippets-admin-views.

There's more work to be done to make snippets a like-for-like replacement for modeladmin, such as allowing them to be registered in arbitrary places in the sidebar - the progress on this is detailed at https://github.com/wagtail/wagtail/discussions/8609. Once this is ready, we'd recommend switching your model over to snippets to benefit from the new functionality.

**Comment by allcaps** [comment]
The modeladmin filters are plain Django admin filters. Which means customizing is documented in the Django docs and not too hard.
